src/Cli/dotnet/CommandDirectoryContext.cs

```diff
@@ -32,14 +32,17 @@ public static void PerformActionWithBasePath(string basePath, Action action)
                 throw new InvalidOperationException(
                     $"Calls to {nameof(CommandDirectoryContext)}.{nameof(PerformActionWithBasePath)} cannot be nested.");
             }
+            var oldPath = Directory.GetCurrentDirectory();
             _basePath = basePath;
+            Directory.SetCurrentDirectory(_basePath);
             Telemetry.Telemetry.CurrentSessionId = null;
             try
             {
                 action();
             }
             finally
             {
+                Directory.SetCurrentDirectory(oldPath);
                 _basePath = null;
```

Comment:
The intent of this helper is to make the app act like it's in a given path, but certain codeflows were using `Directory.GetCurrentDirectory()`. That meant that this helper wasn't doing what it was expected to do. This aligns the helper with its intent.